daemon/loop.go

```diff
@@ -72,13 +50,22 @@ func (d *Daemon) GitPollLoop(stop chan struct{}, wg *sync.WaitGroup, logger log.
 		k(logger)
 	}
 
+	imagePollTimer := time.NewTimer(d.RegistryPollInterval)
+
 	// Ask for a sync, and to poll images, straight away
 	d.askForSync()
+	d.askForImagePoll()
 	for {
 		select {
 		case <-stop:
 			logger.Log("stopping", "true")
 			return
+		case <-d.pollImagesSoon:
+			d.pollForNewImages(logger)
+			imagePollTimer.Stop()
+			imagePollTimer = time.NewTimer(d.RegistryPollInterval)
```

Comment:
Because it can be renewed either by the timer firing or by something asking explicitly for an image poll, and in both those cases we want to reset the timer.